io/roundtrip.go

```diff
@@ -21,8 +22,8 @@ func RoundTripCheck(from any, to func() any) error {
 			if err != nil {
 				return err
 			}
-			if !reflect.DeepEqual(from, r) {
-				return errors.New("reconstructed object don't match original (ReadFrom)")
+			if err = equal(from, r); err != nil {
```

Comment:
I would keep it as is, imo there may be unwanted side-effects of using `Equal` if exists - there are already a few types in gnark which have the Equal method defined and they may be more strict than needed.

It seems like it is used to avoid comparing `Challenge` field in a test. But at least for me it would be clearer to manually empty the Challenge field manually when needed in the test. This current approach may break imo other roundtrip tests.

Reply:
ha, on that one I agree with @ivokub ; I previously had a similar snippet, and had some working round trip tests that actually didn't flag an error because the `Equal` function of an object was not correctly updated in a PR. i.e. `reflect.DeepEqual` is a bit more cumbersome but better for non-regression since it checks everything, including new fields, fields that we precompute, etc, that may not even be checked by a "canonical" `Equal`.